ansible/roles/app_submitty/tasks/main.yml

```diff
@@ -0,0 +1,131 @@
+---
+- name: Run the equivalent of "apt get update" as a seperate stop
+  ansible.builtin.apt:
+    update_cache: yes
+    cache_valid_time: 300
+
+- name: Install PHP
+  ansible.builtin.apt:
+    name: php
+    state: present
+  register: apt_action
+  retries: 100
+  until: apt_action is success or
+    ('Failed to lock apt for exclusive operation' not in apt_action.msg and
+    '/var/lib/dpkg/lock' not in apt_action.msg)
+
+- name: Create submitty git directory if none exists
+  ansible.builtin.file:
+    path: /usr/local/submitty/GIT_CHECKOUT
+    state: directory
+    mode: "0755"
+
+- name: Checkout specific version of the submitty git repo
+  ansible.builtin.git:
+    repo: https://github.com/Submitty/Submitty.git
+    dest: /usr/local/submitty/GIT_CHECKOUT/Submitty
+    version: "{{ submitty_version }}"
+    force: yes
+    update: yes
+
+- name: Run install_system.sh script
+  ansible.builtin.shell: |
+    executable: "/bin/bash"
+    cmd: |
+      set -o pipefail
+      echo  "{{ db_host }}
+        {{ db_port }}
+        "{{ submitty_dbuser }}"
+        "{{ submitty_dbuser_password }}"
+        "{{ submitty_course_dbuser }}"
+        "{{ submitty_course_dbuser_password }}"
+        "{{ timezone }}"
+        "{{ language }}"
+        "{{ 'https' if ssl_enabled else 'http' }}://{{ submitty_url }}"
+        "{{ vcs_url }}"
+        "{{ institution_name }}"
+        "{{ sysadmin_email }}"
+        "{{ submitty_email }}"
+        "{{ institution_url }}"
+        1
+        "{{ admin_email }}"
+        y
+        "{{ admin_email }}"
+        "{{ admin_password }}"
+        "{{ submitty_ip }}"
+        "{{ submitty_email }}"
+        "{{ noreply_email }}"
+        "{{ smtp_server }}"
+        "{{ smtp_port }}"
+        "{{ institution_url }}"
+        " | bash -p ./.setup/install_system.sh > /usr/local/submitty/install.log 2>&1
+
+  args:
+    chdir: /usr/local/submitty/GIT_CHECKOUT/Submitty
+  become: true
+  become_user: root
+
+- name: Check to see if installation succeeded
+  ansible.builtin.shell: grep "INSTALLATION SUCCESS!" /usr/local/submitty/install.log
+
+- name: Update the PHP disabled functions
+  ansible.builtin.lineinfile:
+    path: /etc/php/8.1/fpm/php.ini
+    regexp: "^disable_functions = "
+    line: disable_functions = popen,pclose,proc_open,php_real_logo_guid,php_egg_logo_guid,php_ini_scanned_files,php_ini_loaded_file,readlink,symlink,link,set_file_buffer,proc_close,proc_terminate,proc_get_status,proc_nice,getmyuid,getmygid,getmyinode,putenv,get_current_user,magic_quotes_runtime,set_magic_quotes_runtime,import_request_variables,ini_alter,stream_socket_server,stream_socket_accept,stream_socket_pair,stream_get_transports,stream_wrapper_restore,mb_send_mail,openlog,syslog,closelog,pfsockopen,posix_kill,apache_child_terminate,apache_get_modules,apache_get_version,apache_lookup_uri,apache_reset_timeout,apache_response_headers,virtual,system,phpinfo,exec,shell_exec,passthru,
```

Comment:
It would be good to add a comment with a link to the documentation page that has the list of disabled functions, to make it easier to maintain in the future if this list gets updated.